### DISCUSSION TITLE
Add a crop/delete/merge feature

### DISCUSSION THREAD

**Discussion by tecchese** [sn4k3/UVtools#736] [Ideas]
Hallo, I often need, to save time, to print multiple objects in one session. Of course I can do this in my slicer program but if I don't want to print a part of the sliced files I can't do it.
For example, see the following image:
  
![Immagine 2023-06-27 061327](https://github.com/sn4k3/UVtools/assets/107458958/7ac479ab-1b57-42d7-8e9f-10aa537aa32c)

I don't want to print just the central object, because I don't need it, thus saving some resin. Actually I can't do it (or I don't know/find a tool inside UVTools that can do it), I have to reassemble it first in LycheeSlicer and then export the sliced file with the only objects I need... it's a waste of time , for me. If this option were implemented directly in UVTools it would be great, as well as the possibility to add/remove/duplicate objects with the same print parameters in the same session.
Cheers

**Comment by sn4k3** [comment]
There are at least 4 ways I can think off to remove an object using UVtools.
The easiest which actually is a shortcut of a tool is:

1. Select the area/object to remove with SHIFT (ROI)
2. Then click combination: Alt + Delete

![image](https://github.com/sn4k3/UVtools/assets/113281/a4ce9920-22e3-433f-9f82-8d1bcc8f918c)

___

If you want to keep just the selected object:

1. Select the area/object to remove with SHIFT (ROI)
2. Then click combination: Alt + Insert

https://youtu.be/EpcDmskdaKk

___

There are also multiple ways to duplicate objects, the easiest is by use the Tool - Pattern.
If you want to copy just one from multiple, you can use the ROI + tool Move with the "cut move" unchecked to perform a copy at a location

Note: You should never pattern on slicer if you use UVtools for issue checking and cleaning. It will be much faster for UVtools to check the file, get back to slicer if required and in the end use UVtools to pattern the copies

**Reply by tecchese** [reply]
Thanks, it works...however the lack of an undo makes these operations a little awkward, with a few attempts to do before printing. I will only use the pattern tool to duplicate perfect objects.

**Reply by sn4k3** [reply]
> however the lack of an undo

Ctrl + Z

**Reply by tecchese** [reply]
I think it may be wiser to place the UNDO and REDO buttons (or even CUT, COPY, DELETE, PASTE or other commands) somewhere in the GUI

**Reply by sn4k3** [reply]
Here the buttons, along with all modifications which you can jump and revert to

![image](https://github.com/sn4k3/UVtools/assets/113281/7e079818-c028-44ee-a5ca-92478ea79a48)
